README.md

```diff
@@ -0,0 +1,89 @@
+# mod-sample
+
+Copyright (C) 2019 The Open Library Foundation
+
+This software is distributed under the terms of the Apache License, Version 2.0.
+See the file "[LICENSE](LICENSE)" for more information.
+
+
+## Introduction
+
+FOLIO publisher-subscriber module to provide event-driven approach.
+It is responsible for maintaining the registrations of event types and for coordinating distribution of events to the appropriate subscribers (modules).
+
+## Compiling
+
+```
+   mvn install
+```
+
+See that it says "BUILD SUCCESS" near the end.
+
+## Docker
+
+Build the docker container with:
+
+```
+   docker build -t mod-pubsub.
+```
+
+Test that it runs with:
+
+```
+   docker run -t -i -p 8081:8081 mod-pubsub
+```
+
+## Installing the module
+
+Follow the guide of
+[Deploying Modules](https://github.com/folio-org/okapi/blob/master/doc/guide.md#example-1-deploying-and-using-a-simple-module)
+sections of the Okapi Guide and Reference, which describe the process in detail.
+
+First of all you need a running Okapi instance.
+(Note that [specifying](../README.md#setting-things-up) an explicit 'okapiurl' might be needed.)
+
+```
+   cd .../okapi
+   java -jar okapi-core/target/okapi-core-fat.jar dev
+```
+
+We need to declare the module to Okapi:
+
+```
+curl -w '\n' -X POST -D -   \
+   -H "Content-type: application/json"   \
+   -d @target/ModuleDescriptor.json \
+   http://localhost:9130/_/proxy/modules
+```
+
+That ModuleDescriptor tells Okapi what the module is called, what services it
+provides, and how to deploy it.
+
+## Deploying the module
+
+Next we need to deploy the module. There is a deployment descriptor in
+`target/DeploymentDescriptor.json`. It tells Okapi to start the module on 'localhost'.
+
+Deploy it via Okapi discovery:
+
+```
+curl -w '\n' -D - -s \
+  -X POST \
+  -H "Content-type: application/json" \
+  -d @target/DeploymentDescriptor.json  \
+  http://localhost:9130/_/discovery/modules
+```
+
+Then we need to enable the module for the tenant:
+
+```
+curl -w '\n' -X POST -D -   \
+    -H "Content-type: application/json"   \
+    -d @target/TenantModuleDescriptor.json \
+    http://localhost:9130/_/proxy/tenants/<tenant_name>/modules
+```
+
+## Issue tracker
+
+See project [MODDATAIMP](https://issues.folio.org/browse/MODDATAIMP)
```

Comment:
set link to modpubsub jira project

Reply:
Please add a link to the google document with design for pub-sub (powered by Taras S.)